### DISCUSSION TITLE
How to find proof size?

### DISCUSSION THREAD

**Discussion by sid-alluri** [Consensys/gnark#842] [Q&A]
Is there any tooling available to find proof size in gnark? I am using `reflect.TypeOf(proof).Size()` and the output seems incorrect.

**Comment by gbotrel** [comment]
hey -- you can serialize the `Proof` with `WriteTo` (compressed points) or `WriteTo` (uncompressed) and check the size of the buffer. 
Size is constant for groth16 and plonk and does not depend on the circuit.

**Reply by sid-alluri** [reply]
Does that mean same-size proof regardless of any circuit in Plonk or groth16? I am getting 616B of proof regardless of the circuit. Is this correct?

**Reply by gbotrel** [reply]
plonk proof size != groth16 proof size.

But yes, otherwise it's not circuit dependent (within a proving scheme)

**Reply by sid-alluri** [reply]
Thanks. Got it
